src/core/Logger.cpp

```diff
@@ -110,12 +121,16 @@ LockedLogOutput Logger::log(LogSeverity sev)
     LockedLogOutput out(&m_output, &m_lock);
 
     if(m_color_enabled) {
-        out << ANSI_DARK_GREY
-            << "[" << timetext << "] "
-            << get_severity_color(sev)
-            << sevtext
-            << ": "
-            << ANSI_RESET;
+#ifdef _WIN32
+		GetConsoleScreenBufferInfo(STDOUT, &old_colors);
+#endif
+		out << DARK_GREY;
```

Comment:
The compiler will actually optimize stuff like this better if you let the stream run in one shot, rather than re-calling `<<` multiple times for the same data.

Reply:
I did this on purpose. For some reason, streamlining the output like that doesn't properly update the colors. Instead, it'll update the text to only the first color (DARK_GREY) and stay unchanged, even at RESET. I didn't have time to properly debug it, but I'm suspecting that it's because of some sort of optimization.

Reply:
Looking at a disassembly, the compiler tries to optimize it by evaluating and storing every argument first and then calling the actual `<<` overload after. In order for the text to color properly, `SetConsoleTextAttribute` must be called before each consecutive portion of text and then be written to the console. I also compared running the stream in one shot to re-calling `<<` and there's very little difference in performance.